### PR TITLE
Update docs-app to be more fleshed out and production-ready

### PR DESCRIPTION
**Pull request by ynotdraw**
## 🚀 Description

This PR gets the docs-app in a more publishable state by:

- Updating the `<meta>` `<description>` so that when the link is shared, a more proper description is displayed over what we have today
- Removing the feature cards for now until we figure out what we want to put here
- Adding the CrowdStrike logo now that we are under the Apache v2 license
- Adding a GitHub link to the repo
- Updating the way we specify "Optional" by putting it to the front
- Updating all UI State sections to roughly match
  - **NOTE:** Some components are trickier and have more args than others, so not an exact 1:1
  - **NOTE2:** I did not touch InputFile and Fieldset as those components are getting a few changes and I want to prevent merge conflicts at the moment
- Updating all demos to be more "real"

---

## 🔬 How to Test

- Visit https://ec41bb62.ember-toucan-core.pages.dev
- Verify the CrowdStrike logo in the top-left. Click it and verify it goes to our website.
- Verify the GitHub logo in the top-right.  Click it and verify it goes to this repo.
- Verify the main page no longer has "Coming soon" text
- Verify in the PR that "Optional." is now at the front and replaces "This is optional."
- Browse the UI state demos touched here and verify they're in good shape
  - **NOTE:** Ignore InputFile and Fieldset (see above)

---

## 📸 Images/Videos of Functionality

Images are in the PR when reviewing!

### CODE REVIEW

File: docs-app/app/templates/application.hbs [hunk 1]
```diff
@@ -3,13 +3,29 @@
 <App>
   <:topBar>
     <TopBar>
+      <:logo>
+        <ExternalLink @href="https://crowdstrike.com" class="mt-3">
```
[the remark this answers is not in the document]
Comment (camskene) [reply docs-app/app/templates/application.hbs:7]: The horizontal alignment needs tweaking:

image

[the remark this answers is not in the document]
Comment (ynotdraw) [reply docs-app/app/templates/application.hbs:7]: I'm curious if other folks are seeing this offset as well.  For some reason I can't reproduce the issue on Chrome, Safari, Firefox, or Arc browser 😭 

Screenshot 2023-03-29 at 11 36 52 AM

Screenshot 2023-03-29 at 11 37 33 AM

[the remark this answers is not in the document]
Comment (nicolechung) [reply docs-app/app/templates/application.hbs:7]: This is me in Chrome

Screenshot 2023-03-29 at 11 51 05 AM

File: docs/components/checkbox-field/index.md [hunk 2]
```diff
@@ -103,73 +103,142 @@ Target the error block via `data-error`.
...
+  <Form::CheckboxField
+    @label='This is an option that expands to multiple lines'
+  />
+</div>
 
-<Form::CheckboxField
-@label='Label'
-@hint='With hint text'
-/>
+### CheckboxField with label and hint
```
Comment (nicolechung) [inline docs/components/checkbox-field/index.md:114]: Love how this looks! Will try to update the File Input Field with these headers as well (I'm on two related FileInput PRs, so maybe can put it in one of those)

File: docs/headless-form/changeset/demo/base-demo.md [hunk 3]
```diff
@@ -42,23 +42,27 @@ import {
 } from 'ember-changeset-validations/validators';
 
 export default class MyFormComponent extends Component {
-  data = { name: 'Nicole' };
```
Comment (nicolechung) [inline docs/headless-form/changeset/demo/base-demo.md:45]: ✅

**Review by nicolechung — APPROVED** [review]

File: docs/components/checkbox-field/index.md (hunk 2 above)
Comment (nicolechung) [reply docs/components/checkbox-field/index.md:114]: See: https://github.com/CrowdStrike/ember-toucan-core/pull/110/commits/72b72da48c4c4cd88f9ae7e855376a69f249511b

File: docs/headless-form/native/demo/base-demo.md [hunk 4]
```diff
@@ -8,12 +8,26 @@
...
+
+  <form.Field @name='email' as |field|>
+    <Form::InputField
+      @label='Email'
+      @hint='How do we contact you?'
+      @error={{this.mapErrors field.rawErrors}}
       @value={{field.value}}
       @onChange={{field.setValue}}
+      name='email'
+      pattern='/^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/'
```
[the remark this answers is not in the document]
Comment (simonihmig) [reply docs/headless-form/native/demo/base-demo.md:28]: Any specific reason to use a regex here? You could have used a `type="email"` input, which has built-in email validation.

**Review by simonihmig — APPROVED** [review]
Nice! 😍

File: docs-app/app/templates/application.hbs (hunk 1 above)
[the remark this answers is not in the document]
Comment (simonihmig) [reply docs-app/app/templates/application.hbs:7]: > This is me in Chrome

Same on my Chrome. Cannot reproduce the misalignment... 🤷‍♂️

File: docs/headless-form/native/demo/base-demo.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (ynotdraw) [reply docs/headless-form/native/demo/base-demo.md:28]: Doh! Nope haha. Let me switch to that. I have no idea why I missed this! 😅 thank you!

File: docs-app/app/templates/application.hbs (hunk 1 above)
[the remark this answers is not in the document]
Comment (camskene) [reply docs-app/app/templates/application.hbs:7]: I have the same misalignment in Edge and Firefox :(

[the remark this answers is not in the document]
Comment (nicolechung) [reply docs-app/app/templates/application.hbs:7]: @camskene https://github.com/CrowdStrike/ember-toucan-core/issues/114

**Pull request merged**